### PR TITLE
Add SET-based

### PR DESCRIPTION
**Pull request by ricktu288**
Yi-Ting Tu, quantum.
I think "SET based" may refer to several possible constructions (e.g. anomalous 2D SET boundary for 3D SPT) for the more general symmetry-protected self-correcting quantum memory, so I create the entry for the latter. I also create stubs for two specific models constructed in that paper.

### CODE REVIEW

File: codes/quantum/qubits/subsystem/symmetry_protected_color.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  Stub. (see Sec. IV D of \cite{arXiv:1805.01474})
+
+protection: |
+  The symmetric energy barrier grows linearly with the side length. When the system is coupled locally to a thermal bath respecting the symmetry and below a critical temperature, the memory time grows expoenetially with the side length.
+
+relations:
+  parents:
+    - code_id: symmetry_protected
+      detail: ''
+    - code_id: subsystem_stabilizer
```
Comment (valbert4) [inline codes/quantum/qubits/subsystem/symmetry_protected_color.yml:16]: subsystem_color

File: codes/quantum/properties/symmetry_protected.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+code_id: symmetry_protected
```
Comment (valbert4) [inline codes/quantum/properties/symmetry_protected.yml:1]: - Please add a stub for "Self-correcting quantum code" (code_id: self_correct). 
- Add hamiltonian as parent to self_correct; add higher_dimensional_surface as cousin, mention that it's a self-correcting memory in 4D.
- Add self_correct as parent to this entry; remove hamiltonian.

File: codes/quantum/properties/symmetry_protected.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+name: 'Symmetry-protected self-correcting quantum code'
+introduced: '\cite{arXiv:1805.01474}'
+
+description: |
+  The physical degrees of freedom of the code correspond to a physical lattice with finite-dimensional local Hilbert space. The codespace is the degenerate ground space of a Hamiltonian that consists of terms localized in space. The system is required to have a symmetry not spontaneously broken, with the symmetry-protected self-correcting property described as follows.
+
+  Suppose the physical state is \(\rho(0)\) at time \(t=0\), and is evolved to \(\rho(t)\) by the Hamiltonian plus a small coupling to a thermal bath with a low enough temperature, in the way that is local on the lattice and respects the symmetry. Then the memory time \(\tau_\text{mem}\) is defined to be the maximal \(t\) such that \(\left\|\Phi_\text{ec}(\rho(t))-\rho(0)\right\|_1\leq\epsilon\), where \(\Phi_\text{ec}\) is the error correcting map and \(\epsilon\) a fixed error rate.
```
Comment (valbert4) [inline codes/quantum/properties/symmetry_protected.yml:9]: It'll be good to make the inequality into a display equation; see code_id:shor_nine on how to do that.

File: codes/quantum/properties/symmetry_protected.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
+code_id: symmetry_protected
+
+name: 'Symmetry-protected self-correcting quantum code'
+introduced: '\cite{arXiv:1805.01474}'
+
+description: |
+  The physical degrees of freedom of the code correspond to a physical lattice with finite-dimensional local Hilbert space. The codespace is the degenerate ground space of a Hamiltonian that consists of terms localized in space. The system is required to have a symmetry not spontaneously broken, with the symmetry-protected self-correcting property described as follows.
```
Comment (valbert4) [inline codes/quantum/properties/symmetry_protected.yml:7]: Remember that audience is not always physicists. Need to define what a "symmetry" is here and what does it mean to respect i (i.e., some operator U commuting with Ham). What kinds of U are allowed? You can then use this to help with the text below, like defining what it means for the bath to respect symmetry.

File: codes/quantum/properties/symmetry_protected.yml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+  Suppose the physical state is \(\rho(0)\) at time \(t=0\), and is evolved to \(\rho(t)\) by the Hamiltonian plus a small coupling to a thermal bath with a low enough temperature, in the way that is local on the lattice and respects the symmetry. Then the memory time \(\tau_\text{mem}\) is defined to be the maximal \(t\) such that \(\left\|\Phi_\text{ec}(\rho(t))-\rho(0)\right\|_1\leq\epsilon\), where \(\Phi_\text{ec}\) is the error correcting map and \(\epsilon\) a fixed error rate.
+  \(\tau_\text{mem}\) is required to grow without bound (typically exponentially) in the system size.
+  In addition, the logical operators are required to have a decomposition into a product such that each factor is localized in space and has the symmetry.
+
+  Such a code can be constructed using a 3D symmetry-protected topological (SPT) phase in the bulk with a 1-form symmetry, and an anomalous 2D symmetry-enriched topological (SET) phase on the boundary with ground-state degeneracy, such that the anyons on the boundary are confined by the string-like excitations in the bulk enforced by the symmetry.
```
Comment (valbert4) [inline codes/quantum/properties/symmetry_protected.yml:13]: Careful with audience here; "anyons" "SPT" "SET" "string-like" "confined" all are not defined. It's ambitious to cover all that here. I suggest elaborating that on-site (i.e., tensor-product with weight equal to number of subsystems) is not sufficient, and that 1-form (define...) is.

File: codes/quantum/qubits/cubic_rbh.yml [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+relations:
+  parents:
+    - code_id: symmetry_protected
+      detail: ''
+    - code_id: qubit_stabilizer
+      detail: ''
+  cousins:
+    - code_id: surface
+      detail: 'Without the symmetry protection, the system is effectively a surface code on a face of its boundary.'
```
Comment (valbert4) [inline codes/quantum/qubits/cubic_rbh.yml:20]: Please clarify "on a face of its boundary".

**Comment by valbert4** [comment]
Thanks @ricktu288; please resubmit with changes.

File: codes/quantum/qubits/cubic_rbh.yml (hunk 6 above)
Comment (ricktu288) [reply codes/quantum/qubits/cubic_rbh.yml:20]: I think the name of the face depends on what will be written in the description. Maybe "the front face"? Or "the face with the toric-code boundary condition"? But it is still not defined in this stub.

Comment (valbert4) [reply codes/quantum/qubits/cubic_rbh.yml:20]: "the face with the toric-code boundary condition" sounds OK for now. As long as it's clear there's a way to do BC such that the state (or subspace?) on one face is similar to (or, modulo projecting out the bulk, equal?) to the toric code.

**Pull request closed without merging**